crates/lib/src/jekyll/array.rs

```diff
@@ -10,6 +13,122 @@ use liquid_core::{Value, ValueView};
 
 use crate::invalid_input;
 
+#[derive(Debug, Default, FilterParameters)]
+struct SortArgs {
+    #[parameter(description = "The property accessed by the filter.", arg_type = "str")]
+    property: Option<Expression>,
+    #[parameter(
+        description = "nils appear before or after non-nil values, either ('first' | 'last')",
+        arg_type = "str"
+    )]
+    nils: Option<Expression>,
+}
+
+#[derive(Clone, ParseFilter, FilterReflection)]
+#[filter(
+    name = "sort",
+    description = "Sorts items in an array. The order of the sorted array is case-sensitive.",
+    parameters(SortArgs),
+    parsed(SortFilter)
+)]
+pub struct Sort;
+
+#[derive(Debug, Default, FromFilterParameters, Display_filter)]
+#[name = "sort"]
+struct SortFilter {
+    #[parameters]
+    args: SortArgs,
+}
+
+enum NilsOrder {
+    First,
+    Last,
+}
+
+fn safe_property_getter<'a>(value: &'a Value, property: String) -> Value {
+    let properties = property.split('.').collect::<Vec<&str>>();
+    let mut result = value.to_owned();
+    for property in properties {
+        result = result
+        .as_object()
+        .and_then(move |obj| {
+            if let Some(scalar) = obj.as_scalar() {
+                try_find(&property, &vec![scalar]).map(|v| v.clone().to_value())
```

Comment:
Why are we searching within a property to find the value within the array?

My copy doesn't have this and it passes tests.

In fact, why are we manually looping over `properties` rather than calling `try_find(value, properties)`?

Reply:
The short answer is that I'm not super familiar with the inner workings of this crate. I'll fiddle with this a bit more and update my PR. Thanks for reviewing!
IIRC your implementation did not pass every test (didn't work for nested properties) but I could be misremembering.

Reply:
I meant my updated copy that I referenced in https://github.com/cobalt-org/liquid-rust/pull/546#issuecomment-2143010012

Reply:
Re-read your comment and I see what you mean now. I'll fix this when I get the chance.